aws/resource_aws_redshift_snapshot_schedule_association.go

```diff
@@ -0,0 +1,242 @@
+		ClusterIdentifier:  aws.String(clusterIdentifier),
+		ScheduleIdentifier: aws.String(scheduleIdentifier),
+	}
+
+	resp, err := conn.DescribeSnapshotSchedules(descOpts)
+	if err != nil {
+		return fmt.Errorf("Error describing Redshift Cluster %s Snapshot Schedule %s: %s", clusterIdentifier, clusterIdentifier, err)
+	}
+
+	if resp.SnapshotSchedules == nil || len(resp.SnapshotSchedules) == 0 {
+		return fmt.Errorf("Unable to find Redshift Cluster (%s) Snapshot Schedule (%s) Association", clusterIdentifier, scheduleIdentifier)
+	}
+	snapshotSchedule := resp.SnapshotSchedules[0]
+	if snapshotSchedule.AssociatedClusters == nil || aws.Int64Value(snapshotSchedule.AssociatedClusterCount) == 0 {
+		return fmt.Errorf("Unable to find Redshift Cluster (%s)", clusterIdentifier)
+	}
+
+	var associatedCluster *redshift.ClusterAssociatedToSchedule
+	foundAssociatedCluster := false
+	for _, cluster := range snapshotSchedule.AssociatedClusters {
+		if *cluster.ClusterIdentifier == clusterIdentifier {
+			associatedCluster = cluster
+			foundAssociatedCluster = true
+			break
+		}
+	}
+
+	if !foundAssociatedCluster {
+		return fmt.Errorf("Unable to find Redshift Cluster (%s)", clusterIdentifier)
+	}
+
+	d.Set("cluster_identifier", associatedCluster.ClusterIdentifier)
+	d.Set("schedule_identifier", snapshotSchedule.ScheduleIdentifier)
+
+	return nil
+}
+
+func resourceAwsRedshiftSnapshotScheduleAssociationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).redshiftconn
+	clusterIdentifier, scheduleIdentifier, err := resourceAwsRedshiftSnapshotScheduleAssociationParseId(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parse Redshift Cluster Snapshot Schedule Association ID %s: %s", d.Id(), err)
+	}
+
+	_, err = conn.ModifyClusterSnapshotSchedule(&redshift.ModifyClusterSnapshotScheduleInput{
+		ClusterIdentifier:    aws.String(clusterIdentifier),
+		ScheduleIdentifier:   aws.String(scheduleIdentifier),
+		DisassociateSchedule: aws.Bool(true),
+	})
+
+	if isAWSErr(err, redshift.ErrCodeClusterNotFoundFault, "") {
+		log.Printf("[WARN] Redshift Snapshot Cluster (%s) not found, removing from state", clusterIdentifier)
+		d.SetId("")
+		return nil
+	}
+	if isAWSErr(err, redshift.ErrCodeSnapshotScheduleNotFoundFault, "") {
+		log.Printf("[WARN] Redshift Snapshot Schedule (%s) not found, removing from state", scheduleIdentifier)
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error disassociate Redshift Cluster (%s) and Snapshot Schedule (%s) Association: %s", clusterIdentifier, scheduleIdentifier, err)
+	}
+
+	if err := waitForRedshiftSnapshotScheduleAssociationDestroy(conn, 75*time.Minute, clusterIdentifier, scheduleIdentifier); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsRedshiftSnapshotScheduleAssociationParseId(id string) (clusterIdentifier, scheduleIdentifier string, err error) {
+	parts := strings.SplitN(id, "/", 2)
+	if len(parts) != 2 {
```

Comment:
We should check for empty strings on either side of the `/` as well 👍 

```suggestion
	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
```

Reply:
Thanks, fix it.